srtcore/group.cpp

```diff
@@ -1147,18 +1154,21 @@ int CUDTGroup::sendBroadcast(const char* buf, int len, SRT_MSGCTRL& w_mc)
     // This simply requires the payload to be sent through every socket in the group
     for (gli_t d = m_Group.begin(); d != m_Group.end(); ++d)
     {
-        // Check the socket state prematurely in order not to uselessly
-        // send over a socket that is broken.
-        CUDT* pu = 0;
-        if (d->ps)
-            pu = &d->ps->core();
-
-        if (!pu || pu->m_bBroken)
+        if (d->sndstate != SRT_GST_BROKEN)
         {
-            HLOGC(gslog.Debug,
-                  log << "grp/sendBroadcast: socket @" << d->id << " detected +Broken - transit to BROKEN");
-            d->sndstate = SRT_GST_BROKEN;
-            d->rcvstate = SRT_GST_BROKEN;
+            // Check the socket state prematurely in order not to uselessly
+            // send over a socket that is broken.
+            CUDT* pu = 0;
+            if (d->ps)
+                pu = &d->ps->core();
```

Comment:
Might be better.

```suggestion
            const CUDT* pu = (d->ps)
                ? &d->ps->core()
                : NULL;
```

Reply:
I believe you meant `CUDT* const pu = (d->ps) ... `

Reply:
Well, both. `pu` is not modified here, so `const CUDT* const pu =...`.

Reply:
Right. In practice actually, this intermediate variable is only for having a shortcut - the condition would be more elaborate without it.